File: backend/app/schemas/analytics.py
from app.schemas.common import ORMModel


class AnalyticsOverview(ORMModel):
    total_incidents: int
    open_incidents: int
    critical_risks: int
    open_permits: int
    overdue_maintenance: int
    equipment_health_average: float
    incident_severity_breakdown: dict[str, int]
    risk_severity_breakdown: dict[str, int]
    equipment_status_breakdown: dict[str, int]
    permit_status_breakdown: dict[str, int]
    maintenance_status_breakdown: dict[str, int]
    department_safety_scores: dict[str, float]

File: frontend/src/types/analytics.ts
export interface AnalyticsOverview {
  total_incidents: number;
  open_incidents: number;
  critical_risks: number;
  open_permits: number;
  overdue_maintenance: number;
  equipment_health_average: number;
  incident_severity_breakdown: Record<string, number>;
  risk_severity_breakdown: Record<string, number>;
  equipment_status_breakdown: Record<string, number>;
  permit_status_breakdown: Record<string, number>;
  maintenance_status_breakdown: Record<string, number>;
  department_safety_scores: Record<string, number>;
}

